app/src/main/java/openfoodfacts/github/scrachx/openfood/features/product/edit/ProductEditActivity.kt

```diff
@@ -527,7 +528,7 @@ class ProductEditActivity : BaseActivity() {
             try {
                 productsApi.performOCR(code, imageField).await()
             } catch (err: Exception) {
-                ingredientsFragment.hideOCRProgress()
+                MainScope().launch { ingredientsFragment.hideOCRProgress() }
```

Comment:
I think `withContext` operator is better here
`withContext(Dispatchers.MAIN) { ingredientsFragment.hideOCRProgress() }`

Reply:
Not really sure what crash you can get from this. And how is `withContext ` better for the reason you mentioned.

Reply:
`withContext` runs within the `lifecycleScope` (line 410) and the job will be canceled when the activity gets destroyed.
MainScope doesn't know about activity lifecycle and you have to cancel it when activity destroys. In this part of code you create a new job, which will run in scope, which is not lifecycle aware and can crash because `ingredientsFragment` not exists anymore.
I understand a very small chance that this will happen but this is not a good reason to use MainScope here.

Reply:
You are aware the changes in the suspend block are for a fragment in another `lifecycleScope` ?

Reply:
Yes, and this fragment is attached to this activity. What's the question?

Reply:
Nevermind. :)